src/matches.rs

```diff
@@ -292,7 +301,8 @@ fn block_can_be_flattened<'a>(
         ast::ExprKind::Block(ref block, _)
             if !is_unsafe_block(block)
                 && !context.inside_macro()
-                && is_simple_block(context, block, Some(&expr.attrs)) =>
+                && is_simple_block(context, block, Some(&expr.attrs))
+                && !stmt_is_expr_mac(&block.stmts[0]) =>
```

Comment:
Apparently we haven't backported support for comments in arm guards to v1.x which is why this was removed in 7c24c1e643232ff55f217014787742020382a830